File: server/OurCity.Api/Infrastructure/Database/Comment.cs
namespace OurCity.Api.Infrastructure.Database;

public class Comment
{
    public Guid Id { get; set; }

    public required Guid PostId { get; set; }

    public required Guid AuthorId { get; set; }

    public required string Content { get; set; }

    public required bool IsDeleted { get; set; } = false;

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    // Navigation Properties
    public Post? Post { get; set; }

    public User? Author { get; set; }

    public required List<CommentVote> Votes { get; set; } = new();
}

File: webapp/src/models/comment.ts
/*
  /webapp/src/models/comment.ts

  This file contains the Comment domain model, which represents the Comment entity in the application.
*/
import { VoteType } from "../types/enums";

// Post model representing the Post entity in the application
export interface Comment {
  id: string;
  postId: string;
  authorId: string;
  content: string;
  authorName?: string;
  upvoteCount: number;
  downvoteCount: number;
  voteCount: number; // total votes (upvotes - downvotes)
  voteStatus: VoteType; // UPVOTE = 1| DOWNVOTE = -1| NONE = 0
  isDeleted: boolean;
  createdAt: Date;
  updatedAt: Date;
}